packages/skytrace/src/commands/ping.ts

```diff
@@ -433,31 +433,44 @@ class PingCommand extends Command {
       const isJSON = contentType.match(/json/gi);
       const isXML = contentType.match(/html/gi) || contentType.match(/xml/gi);
 
-      if (flags.showBody) {
-        let language;
-        if (isJSON) {
-          language = 'json';
-        }
-        if (isXML) {
-          language = 'html';
+      let body = context.vars.body;
+      let isBodyValidJSON = false;
+      if (isJSON) {
+        try {
+          body = JSON.parse(context.vars.body);
+          isBodyValidJSON = true;
+        } catch (_parseErr) {
         }
+      }
 
-        if (language) {
-          let output = context.vars.body;
-          if (language === 'json' && flags.pretty) {
-            output = JSON.stringify(JSON.parse(context.vars.body), null, 4);
-          }
-          this.log(highlight(output, { language }));
-        } else {
+      // If -b is set without -p -- just print the body as-is.
+      // If -bp is set - pretty-print JSON or XML/HTML, otherwise print the body as-is.
+      if (flags.showBody) {
+        if (!flags.pretty) {
           this.log(context.vars.body);
+        } else {
+          if (isJSON) {
+            if (!isBodyValidJSON) {
+              this.log(chalk.red('Could not parse body is valid JSON for pretty-printing'));
+            } else {
+              this.log(highlight(JSON.stringify(body, null, 4), { language: 'json' }));
+            }
+
+            if (isXML) {
+              this.log(highlight(context.vars.body, { language: 'html' }));
+            }
+
+            if (!isJSON && !isXML) {
+              this.log(context.vars.body);
+            }
```

Comment:
shouldn't this code path be outside the `if(isJson){}` statement? I don't think this ever gets run otherwise?

Reply:
yep good catch!